draft-ietf-quic-http.md

```diff
@@ -321,8 +321,19 @@ defined in this document: control streams ({{control-streams}}) and push streams
 ({{push-streams}}).  Other stream types can be defined by extensions to HTTP/3;
 see {{extensions}} for more details.
 
-Both clients and servers SHOULD send a value of three or greater for the QUIC
-transport parameter `initial_max_uni_streams`.
+The performance of HTTP/3 connections in the early phase of their lifetime is
+sensitive to the creation and exchange of data on unidirectional streams.
+Endpoints that set low values for the QUIC transport parameters
+`initial_max_uni_streams` and `initial_max_stream_data_uni` will increase the
+chance that the remote peer reaches the limit early. In particular, the value
+chosen for `initial_max_uni_streams` should consider that remote peers may wish
+to exercise reserved stream behaviour ({{stream-grease}}). Limits can be
+increased post handshake at the cost of additional RTT, for example by
+exchanging `STREAMS_BLOCKED` and `MAX_STREAMS` QUIC frames. To reduce the
+possibility of this situation occuring, both clients and servers SHOULD send a
+value of three or greater for the QUIC transport parameter
+`initial_max_uni_streams`, and a value of eight or greater for the QUIC
+transport `initial_max_stream_data_uni`.
```

Comment:
While I understand the motivation to suggest something for `initial_max_stream_data_uni`, 8 bytes sounds strangely small.

I'd argue that we should recommend a value that is greater than the size of SETTINGS frames that most endpoints would send. Also, considering the fact that QUIC endpoint needs to be capable of processing at least one full-sized packet at a time, I do not think that even the most restricted endpoints would set `initial_max_stream_data_uni` below one MTU.

Therefore, I might propose something like "SHOULD be at least 1,024 bytes."

Reply:
8 bytes gives just enough data to see what the stream type is before committing to receiving any further data. However, I also agree that such a low value in practice will not fix the issues around blocking, so I'll accept your suggestion.